### PR TITLE
Bundle miner binding

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: src/interface.cpp [hunk 1]
```diff
@@ -264,6 +266,53 @@ static NAN_METHOD(transactionHash) {
...
+  }
+
+  byte_t bundleNormalizedMax[81];
+  v8::Local<v8::Array> bundle_array = v8::Local<v8::Array>::Cast(info[0]);
+  for (size_t i = 0; i < bundle_array->Length(); i++) {
+    bundleNormalizedMax[i] =
+        bundle_array->Get(i)->NumberValue(Nan::GetCurrentContext()).FromJust();
+  }
+
+  uint64_t security =
```
Comment (rajivshah3) [inline src/interface.cpp]:
```suggestion
  uint8_t security =
```

File: src/interface.cpp [hunk 2]
```diff
@@ -264,6 +266,53 @@ static NAN_METHOD(transactionHash) {
...
+      static_cast<uint64_t>(Nan::To<unsigned>(info[1]).FromJust());
```
Comment (rajivshah3) [inline src/interface.cpp]:
```suggestion
      static_cast<uint8_t>(Nan::To<unsigned>(info[1]).FromJust());
```

File: src/interface.cpp [hunk 3]
```diff
@@ -264,6 +266,53 @@ static NAN_METHOD(transactionHash) {
...
+  uint64_t essenceLength =
```
Comment (rajivshah3) [inline src/interface.cpp]:
```suggestion
  size_t essenceLength =
```

File: src/interface.cpp [hunk 4]
```diff
@@ -264,6 +266,53 @@ static NAN_METHOD(transactionHash) {
...
+      static_cast<uint64_t>(Nan::To<unsigned>(info[3]).FromJust());
```
Comment (rajivshah3) [inline src/interface.cpp]:
```suggestion
      static_cast<size_t>(Nan::To<unsigned>(info[3]).FromJust());
```

File: src/interface.cpp [hunk 5]
```diff
@@ -264,6 +266,53 @@ static NAN_METHOD(transactionHash) {
...
+  trit_t *essence = (trit_t *)malloc(sizeof(trit_t) * essenceLength);
+  v8::Local<v8::Array> essence_array = v8::Local<v8::Array>::Cast(info[2]);
+  for (size_t i = 0; i < essence_array->Length(); i++) {
+    essence[i] =
+        essence_array->Get(i)->NumberValue(Nan::GetCurrentContext()).FromJust();
+  }
+  uint64_t count = static_cast<uint64_t>(Nan::To<unsigned>(info[4]).FromJust());
```
Comment (rajivshah3) [inline src/interface.cpp]:
```suggestion
  uint32_t count = static_cast<uint32_t>(Nan::To<unsigned>(info[4]).FromJust());
```

File: src/interface.cpp [hunk 6]
```diff
@@ -264,6 +266,53 @@ static NAN_METHOD(transactionHash) {
...
+  uint64_t nprocs =
```
Comment (rajivshah3) [inline src/interface.cpp]:
```suggestion
  uint8_t nprocs =
```

File: src/interface.cpp [hunk 7]
```diff
@@ -264,6 +266,53 @@ static NAN_METHOD(transactionHash) {
...
+      static_cast<uint64_t>(Nan::To<unsigned>(info[5]).FromJust());
```
Comment (rajivshah3) [inline src/interface.cpp]:
```suggestion
      static_cast<uint8_t>(Nan::To<unsigned>(info[5]).FromJust());
```

File: src/interface.cpp [hunk 8]
```diff
@@ -264,6 +266,53 @@ static NAN_METHOD(transactionHash) {
...
+
+  if (bundle_miner_mine(bundleNormalizedMax, security, essence, essenceLength,
+                        count, nprocs, &index) != RC_OK) {
+    info.GetReturnValue().Set(-1);
+    free(essence);
+    Nan::ThrowError("Bundle mining failed");
```
Comment (rajivshah3) [inline src/interface.cpp]: If an error is thrown, will we still receive `-1` as the index?

Comment (thibault-martinez) [reply src/interface.cpp]: No idea :)

Comment (rajivshah3) [reply src/interface.cpp]: I wonder if we should just return `-1` and not throw an error since that's what we've done for Android and iOS

**Pull request merged**